### ISSUE TITLE
PATCH for pipeline is akward

### ISSUE THREAD

**Issue by gz** [feldera/feldera#2101]
You tell me if it's a bug, but I find it awkward that I need to specify the name and program name every time I send a patch request to the pipeline, e.g., I want to set connectors I still have to specify the other 3

```
curl -L -X PATCH 'http://localhost:8080/v0/pipelines/supply_chain' \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-d '{
  "connectors": [
    {
      "connector_name": "vendor-datagen",
      "is_input": true,
      "name": "vendor-to-vendor-datagen",
      "relation_name": "vendor"
    }
  ],
  "name": "supply_chain",
  "description": "",
  "program_name": "supply_chain"
}'
```

shouldn't they just be left unchanged in case I don't specify them? I already give the name in the URL

**Comment by lalithsuresh** [comment]
@gz this is 100% a bug inherited from the early prototype -- I think there's even a comment in the code that the name/desc etc shouldn't be optional for PATCH. We never got around to cleaning it up, but it will probably be handled in the new API.

**Comment by gz** [comment]
with the new API we could also drop description?

**Comment by lalithsuresh** [comment]
Good point. We don't actually display description anywhere. It's harmless to keep, I think.

**Comment by blp** [comment]
Oh, this would be great. I have an awful script to turn on cpu profiling on a pipeline (it's obsolete with the new API, I'm sure) and that would have simplified it a lot.

**Comment by snkas** [comment]
New API has clear PATCH semantics.
